Tasks/DeployAzureResourceGroup/Utility.ps1

```diff
@@ -752,4 +755,154 @@ function Get-AzureRMVMsConnectionDetailsInResourceGroup
+
+    if($customScriptExtension)
+    {
+        $subStatuses = $customScriptExtension.SubStatuses
+        $subStatusesStr = $subStatuses | Out-String
+
+        Write-Verbose -Verbose "Custom script extension execution statuses: $subStatusesStr"
+
+        if($subStatuses)
+        {
+            foreach($subStatus in $subStatuses)
+            {
+                if($subStatus.Code.Contains("ComponentStatus/StdErr") -and (-not [string]::IsNullOrEmpty($subStatus.Message)))
+                {
+                    $isScriptExecutionPassed = $false
+                    $errMessage = $subStatus.Message
+                    break
+                }
+            }
+        }
+        else
+        {
+            $isScriptExecutionPassed = $false
+            $errMessage = "No execution status exists for the custom script extension '$extensionName'"
+        }
+    }
+    else
+    {
+        $isScriptExecutionPassed = $false
+        $errMessage = "No custom script extension '$extensionName' exists"     
+    }
+
+    if(-not $isScriptExecutionPassed)
+    {
+        Write-TaskSpecificTelemetry "ENABLEWINRM_ExecutionOfVmCustomScriptFailed"
+        throw (Get-LocalizedString -Key "Failed to enable remote deployment. Setting the custom script extension '{0}' for virtual machine '{1}' failed with error : {2}" -ArgumentList $extensionName, $vmName, $errMessage)
+    }
+
+    Write-Verbose -Verbose "Validated the script execution successfully"
+}
+
+function Is-WinRMCustomScriptExtensionExists
+{
+    param([string]$resourceGroupName,
+    [string]$vmName,
+    [string]$extensionName)
+	 
+    $isExtensionExists = $true
+    $removeExtension = $false
+	
+    try
+    {
+        $customScriptExtension = Get-AzureMachineCustomScriptExtension -resourceGroupName $resourceGroupName -vmName $vmName -name $extensionName -ErrorAction SilentlyContinue					
+        if($customScriptExtension.ProvisioningState -ne "Succeeded")
+        {	
+            $removeExtension = $true		    
+        }
+        else
+        {
+            try
+            {
+                Validate-CustomScriptExecutionStatus -resourceGroupName $resourceGroupName -vmName $vmName -extensionName $extensionName -Verbose
+            }
+            catch
+            {
+                $removeExtension = $true
+            }				
+        }
+    }
+    catch
+    {
+        $isExtensionExists = $false	
+    }		
+    
+    if($removeExtension)
+    {
+        $response = Remove-AzureMachineCustomScriptExtension -resourceGroupName $resourceGroupName -vmName $vmName -name $extensionName -Verbose
+        $isExtensionExists = $false
+    }
+	
+    $isExtensionExists
+}
+
+function Add-AzureVMCustomScriptExtension
+{
+    param([string]$resourceGroupName,
+          [object]$vmResource)
+	
+    $vmName = $vmResource.Name
+    $location=$vmResource.Location	
+    $dnsName=[string]::Concat("*.",$location,".cloudapp.azure.com")
```

Comment:
Should we hard code the suffix? Is it possible to get that from resource property?